app/src/controllers/object.js

```diff
@@ -253,6 +293,45 @@ const controller = {
     }
   },
 
+  /**
+     * @function deleteTags
+     * Deletes the tag set on the requested object
+     * @param {object} req Express request object
+     * @param {object} res Express response object
+     * @param {function} next The next callback function
+     * @returns {function} Express middleware function
+    */
+  async deleteTags(req, res, next) {
+    try {
+      const objId = addDashesToUuid(req.params.objId);
+      const objPath = getPath(objId);
+      const { versionId } = req.query;
+      const objectTagging = await storageService.getObjectTagging({ filePath: objPath, versionId });
+
+      // Generate object subset by subtracting/omitting defined keys via filter/inclusion
+      const keysToRemove = mixedQueryToArray(req.query.key);
+      let newTags = undefined;
+      if (keysToRemove && objectTagging.TagSet) {
+        newTags = objectTagging.TagSet.filter(x => !keysToRemove.includes(x.Key));
+      }
+
+      const data = {
+        filePath: objPath,
+        tags: newTags,
+        versionId: versionId ? versionId.toString() : undefined
+      };
+
+      let response = await storageService.deleteObjectTagging(data);
+      if (newTags) {
+        response = await storageService.putObjectTagging(data);
```

Comment:
This section seems dangerous as we're potentially overwriting the data coming from deleteObjectTagging. Should it be an either or situation instead?